src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java

```diff
@@ -519,6 +521,13 @@ private Entity createEntity(EntityType type, Vector3d position, boolean naturall
         return Optional.empty();
     }
 
+    @Inject(method = "onEntityRemoved", at = @At(value = "HEAD"))
+    public void onEntityRemoval(net.minecraft.entity.Entity entityIn, CallbackInfo ci) {
+        if (!this.isRemote && (!(entityIn instanceof EntityLivingBase) || entityIn instanceof EntityArmorStand)) {
```

Comment:
The `instanceof` checks are also in `callDestructEntityEvent`, you don't need them in both methods.